library/core/tests/num/midpoint.rs

```diff
@@ -0,0 +1,54 @@
+//! Test the following expectations:
+//!  - midpoint(a, b) == (a + b) / 2
+//!  - midpoint(a, b) == midpoint(b, a)
+//!  - midpoint(-a, -b) == -midpoint(a, b)
+
+#[test]
+#[cfg(not(miri))]
```

Comment:
oh I forgot to mention that! yes, this runs fast on even a slow "real" processor, but is unbearable on MIRI. :P